components/core/src/clp/clo/CommandLineArguments.cpp

```diff
@@ -80,18 +80,25 @@ CommandLineArguments::parse_arguments(int argc, char const* argv[]) {
             "Ignore case distinctions in both WILDCARD STRING and the input files"
     );
 
-    po::options_description options_batch_control("Batch Controls");
-    options_batch_control.add_options()(
+    po::options_description options_results_control("Results Controls");
+    // clang-format off
+    options_results_control.add_options()(
             "batch-size,b",
             po::value<uint64_t>(&m_batch_size)->value_name("SIZE")->default_value(m_batch_size),
-            "Batch size"
+            "Batch size for sending results to MongoDB"
+    )(
+            "num-latest-results,n",
```

Comment:
If that's the case, we can get rid of some `num_latest_results == 0` branches. What's the default value of max_results?

Reply:
I was thinking we could leave `num_latest_results == 0` as meaning there's no limit, but I'm also fine if we don't do that. We could set the default to 1k?

Reply:
Yeah, maybe we should support no limit

Reply:
Or should we use UINT64_MAX as for any cases we need to sort the results?

Reply:
You mean should we make `UINT64_MAX` mean "emit all the results in sorted order"? I think we can leave sorting as another flag in the future. Right now, although we emit the results in sorted order, we don't actually need to since they will be sorted later; so until we explicitly need it, it's probably better that we don't say we support it.

Reply:
I'm a little bit confused. As mentioned before, we always return the latest `max-results` results, how can we return latest results without sorting? Or do you mean if `num_results == 0`, we don't sort the results and return all of them; if `num_results > 0`, we sort the results and return at most `num_results` results? If that's the case, how can the sorting flag control `num_results > 0`

Reply:
> Or do you mean if num_results == 0, we don't sort the results and return all of them; if num_results > 0, we sort the results and return at most num_results results?

Yes.

> If that's the case, how can the sorting flag control num_results > 0

The sort flag I'm talking about only controls whether the output is sorted, not whether we sort internally. Whether we sort internally is an implementation detail. We need to sort in the general case, but there might be an optimization where we don't bother sorting if the query doesn't find at least `num_results`.